pkg/transport/generic_transport_controller.go

```diff
@@ -416,46 +580,225 @@ func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, bind
 	return objectsToPropagate, nil
 }
 
-func (c *genericTransportController) initializeWrappedObject(ctx context.Context, binding *v1alpha1.Binding) (*unstructured.Unstructured, error) {
+func (c *genericTransportController) initializeWrappedObject(ctx context.Context, binding *v1alpha1.Binding) (func(v1alpha1.Destination) (*unstructured.Unstructured, bool), []string, error) {
```

Comment:
This function is too packed and hard to trace.
Can you outline its functional flow in compact description, while splitting it into smaller functions that separate concerns? I think this would not only make this function clearer, but as it seems to be a central function, its increased clarity would boost understanding the rest of the code.

For example, if I didn't miss any functionality in the flow:
1. `getObjectsToPropagate`: retrieves the objects from WDS based on the binding
2. `processObjectsForCustomization`: examines each object to determine if it needs customization based on its annotations, and processes accordingly
3. (already in code) `customizeForDestination `: performs the customization for an individual object
4. `appendObjectToDestination`: appends an object to the corresponding destination in the destToCustomizedObjects map while ensuring that objects are correctly grouped by their destination
5. `updateBindingSensitivities`: updates the binding sensitivities based on the destinations involved in the customization
6. `prepareWrappedObjects`: prepares the final wrapped objects for each destination
7. finally, `wrapObjects`: converts objects into their unstructured form

This is the flow I could pick up from the function. Anything that achieves separation of concerns and makes the function easier to process is ok.